src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/MemoryBlocks/StreamMemoryBlockProvider.cs

```diff
@@ -101,29 +101,29 @@ protected override AbstractMemoryBlock GetMemoryBlockImpl(int start, int size)
 
             if (_useMemoryMap && size > MemoryMapThreshold)
             {
-                MemoryMappedFileBlock block;
+                MemoryMappedFileBlock? block;
                 if (TryCreateMemoryMappedFileBlock(absoluteStart, size, out block))
                 {
-                    return block;
+                    return block!;
```

Comment:
This is the case we should use `NotNullWhen` attribute for out parameter. Then we will not need to add bang here 
```csharp
bool TryCreateMemoryMappedFileBlock(long start, int size, [NotNullWhen(true)] out MemoryMappedFileBlock? block)
```
From guidelines: 

> DO annotate non-generic out reference type arguments on Try methods as being nullable and with [NotNullWhen(true)] (e.g. TryCreate([NotNullWhen(true)] out Semaphore? semaphore)). For non-generic arguments, the argument should be nullable because a failed call will generally store null into the argument, and it should be [NotNullWhen(true)] because a successful call will store non-null.